motr/io_nw_xfer.c

```diff
@@ -873,9 +879,32 @@ static int target_ioreq_iofops_prepare(struct target_ioreq *ti,
 				else
 					buf = bvec->ov_buf[seg];
 
-				rc = m0_rpc_bulk_buf_databuf_add(rbuf,
-					buf, COUNT(ivec, seg),
-					INDEX(ivec, seg), ndom);
+				xfer_len = COUNT(ivec, seg);
+				offset = INDEX(ivec, seg);
+				
+				/*
+				 * Accomodate multiple pages in a single 
+				 * net buffer segment if they are consecutive
+				 * pages
+				 */
+				while (seg < SEG_NR(ivec) && 
+				       xfer_len < max_seg_size) {
+					if (filter == PA_DATA && 
+					    read_in_write && auxbvec != NULL &&
+					    auxbvec->ov_buf[seg] != NULL)
+						bufnext = auxbvec->ov_buf[seg+1];
+					else
+						bufnext = bvec->ov_buf[seg+1];
+					if ((m0_bcount_t)buf + xfer_len == 
+					    (m0_bcount_t)bufnext)
+						xfer_len += COUNT(ivec, ++seg);
```

Comment:
Please make sure,
```
auxbvec->ov_buf[seg+1]
bvec->ov_buf[seg+1];
COUNT(ivec, ++seg);
```
are valid even when "seg == SEG_NR(ivec) - 1".

Is there a possibility of overflow?

Reply:
Corrected